### DISCUSSION TITLE
SMR-F Files

### DISCUSSION THREAD

**Discussion by MasterCodeIT** [jglim/CaesarSuite#34] [General]
love your cff splice tool.

do you have something similar or something that can decrypt MB smr-f files to extract the .bin?

**Comment by jglim** [comment]
No, most of the time the matching CFF should be available with the SMR-F so this hasn't been an issue yet. 

The closest project would be [ODB](https://github.com/jglim/ODB), which at its current state, is capable of stripping the protections from the SMR container. However it is a library, not an user-facing tool.

Through ODB, the [raw flash segment data](https://github.com/jglim/ODB/blob/2a0029e68eee3df92ff400ecb98824d2e8d1d32e/ODB/ObjectDB/ODBFile.cs#L60) is available, but there's an additional header that requires further parsing to transform it into a meaningful block. Furthermore, the ODB content has to be parsed to understand the flash segment's address map.

**Comment by bobbb92** [comment]
Hi sir 
Your ODB link is broken. Could you please provide it again?

**Comment by rayanke** [comment]
Anyone still has jglim's ODB extractor tool? Can you kindly share it my telegram: l0wk3yz

**Comment by mgeguren** [comment]
I'm also looking for that smr-f decompile tool.

**Comment by automech** [comment]
also looking for the ODB tool
